src/util/screensaver.cpp

```diff
@@ -0,0 +1,304 @@
+        QDBusInterface iface(SCREENSAVERS[i][0], SCREENSAVERS[i][1], SCREENSAVERS[i][2], 
+            QDBusConnection::sessionBus());
+        if (iface.isValid()) {
+            QDBusReply<uint32_t> reply = iface.call(SCREENSAVERS[i][3], "org.mixxxdj","Mixxx active");
+            if (reply.isValid()) {
+                s_cookie = reply.value();
+                s_saverindex = i;
+                s_enabled = true;
+                qDebug() << "DBus screensaver " << SCREENSAVERS[i][0] <<" inhibited";
+                break;
+            } else {
+                qWarning() << "Call to inhibit for " << SCREENSAVERS[i][0] << " failed: " 
+                    << reply.error().message();
+            }
+        } else {
+            qDebug() << "DBus interface " << SCREENSAVERS[i][0] << " not valid";
+        }
+    }
+}
+void ScreenSaverHelper::uninhibitInternal()
+{
+    if (s_cookie > 0) {
+        s_enabled = false;
+        QDBusInterface iface(SCREENSAVERS[s_saverindex][0], SCREENSAVERS[s_saverindex][1], 
+            SCREENSAVERS[s_saverindex][2],  QDBusConnection::sessionBus());
+        if (iface.isValid()) {
+            QDBusReply<void> reply = iface.call("UnInhibit", s_cookie);
+            if (reply.isValid()) {
+                s_cookie = 0;
+                qDebug() << "DBus screensaver " << SCREENSAVERS[s_saverindex][0] << " uninhibited";
+            } else {
+                qWarning() << "Call to uninhibit for " << SCREENSAVERS[s_saverindex][0] << " failed: " 
+                    << reply.error().message();
+            }
+        } else {
+            qDebug() << "DBus interface " << SCREENSAVERS[s_saverindex][0] << " not valid";
+        }
+    }
+}
+
+#elif HAS_XWINDOW_SCREENSAVER
+// This is untested.
+struct LibXtst : public library {
+  function<int (Display*, unsigned int, Bool, unsigned long)> XTestFakeKeyEvent;
+
+  LibXtst() {
+    if(open("Xtst")) {
+      XTestFakeKeyEvent = sym("XTestFakeKeyEvent");
+    }
+  }
+} libXtst;
+
+void ScreenSaverHelper::triggerUserActivity()
+{
+    char *name = ":0.0";
+    Display *display;
+    if (getenv("DISPLAY"))
+        name=getenv("DISPLAY");
+    display=XOpenDisplay(name);
+    libXtst.XTestFakeKeyEvent(display, 255, True,  0);
+    libXtst.XTestFakeKeyEvent(display, 255, False, 0);
+    XCloseDisplay(display);
+}
+void ScreenSaverHelper::inhibitInternal()
+{
+    char *name = ":0.0";
+    Display *display;
+    if (getenv("DISPLAY"))
+        name=getenv("DISPLAY");
+    display=XOpenDisplay(name);
+    XScreenSaverSuspend(display,True);
+    s_enabled = true;
+}
+void ScreenSaverHelper::uninhibitInternal()
+{
+    char *name = ":0.0";
+    Display *display;
+    if (getenv("DISPLAY"))
+        name=getenv("DISPLAY");
+    display=XOpenDisplay(name);
+    XScreenSaverSuspend(display, False);
+    s_enabled = false;
+}
+
+#else
+void ScreenSaverHelper::inhibitInternal()
+{
+    qError("Screensaver suspending not implemented");
```

Comment:
My idea was, that it is an error, but only one the developers should bother, that`s why I have proposed a DEBUG_ASSERT. 
An 
```#error not implemented``` 
will work even better. 
I am nor sure if it is worth to spend extra work on this unlikely case. 
Fixing the log spam issue by a  boolean will work for me as well.